Source/WebCore/page/InteractionRegion.cpp

```diff
@@ -60,6 +62,20 @@ static CursorType cursorTypeForElement(Element& element)
     return cursorType;
 }
 
+static bool skipNonFullscreenElements(const RenderObject& regionRenderer)
+{
+    auto* fullscreenNode = regionRenderer.document().fullscreenManager().currentFullscreenElement();
+    if (!fullscreenNode)
+        return false;
+
+    for (auto* node = regionRenderer.node(); node; node = node->parentNode()) {
```

Comment:
Probably could use one of the ElementAncestorIterator functions here instead (doubly so since `currentFullscreenElement` will only give you Elements). But also I expect we already have a "is an ancestor of" function somewhere.

Beyond that, though, it seems unfortunate to walk back up the tree for every region.

Also noticed the `ContainsFullScreenElement` Node bit, but that's not quite what you want (you're looking for "is child of full screen element", right?), sadly.

You should ask around! Someone might have a good idea.

Reply:
oh maybe I could use `ContainsFullScreenElement` if I stop the building of interaction regions at a higher level rather than from this function